### ISSUE TITLE
LSP: Error getting actions from TypeScript - error code: -32603

### ISSUE THREAD

**Issue by c4spar** [denoland/deno#9715]
I am getting the following lsp error in my project when I comment out the below code in vscode.

```
Starting Deno language server...
  version: 1.8.0 (release, x86_64-apple-darwin)
Connected to "Visual Studio Code" 1.54.1
language server initialized
Server ready.
Error getting actions from TypeScript: TypeError: Cannot read property 'kind' of undefined
    at Object.isImportClause (deno:cli/tsc/00_typescript.js:27154:21)
    at Object.deleteDeclaration (deno:cli/tsc/00_typescript.js:135890:32)
    at _loop_9 (deno:cli/tsc/00_typescript.js:135398:47)
    at ChangeTracker.finishDeleteDeclarations (deno:cli/tsc/00_typescript.js:135405:21)
    at ChangeTracker.getChanges (deno:cli/tsc/00_typescript.js:135425:22)
    at Function.ChangeTracker.with (deno:cli/tsc/00_typescript.js:134880:32)
    at Object.getCodeActions (deno:cli/tsc/00_typescript.js:140773:69)
    at deno:cli/tsc/00_typescript.js:136053:77
    at Object.flatMap (deno:cli/tsc/00_typescript.js:709:25)
    at Object.getFixes (deno:cli/tsc/00_typescript.js:136053:23)
[Error - 13:47:06] Request textDocument/codeAction failed.
  Message: Internal error
  Code: -32603
```

I don't know exactly what is causing this issue, but it occurs when I comment out the following code. I can't reproduce it with a minimal example. Maybe it has something to do with generic types.

```
type InferCommandWithParent<C extends Command, P extends Command> = C extends
  Command<infer O, infer A, infer G, infer PG, any> ? Command<O, A, G, PG, P>
  : never;
```

**Comment by wiremoons** [comment]
Having same issue with Deno v 1.9.2 on Windows 10 running VSC version: 1.55.2

When I edit comments (adding new ones mostly - see copy below) I get a stream of errors from the 'Deno Language Server' output in VSC as below:
```
Error processing TypeScript request: TypeError: Cannot read property 'kind' of undefined
    at Object.getImplementationsAtPosition (deno:cli/tsc/00_typescript.js:123857:29)
    at Object.getImplementationAtPosition (deno:cli/tsc/00_typescript.js:149882:41)
    at serverRequest (deno:cli/tsc/99_main_compiler.js:676:27)
    at [native_code]:1:12
[Error - 07:32:39] Request codeLens/resolve failed.
  Message: Internal error
  Code: -32603 
```

Example comments being edited that cause this are:
```
#!/usr/bin/env -S deno run --allow-net=api.spacexdata.com
//
//  Run from the command line with command:
//  deno run --allow-net=api.spacexdata.com ./sxl.ts
//
//  Main API Site: https://github.com/r-spacex/SpaceX-API
//
//  Next scheduled launch URL:
//  https://api.spacexdata.com/v4/launches/next
//
//  Payload data URL:
//  https://api.spacexdata.com/v4/payloads/:id
//
```

Is there anyway to turn these error popups off until it is fixed?

Thanks

**Comment by kitsonk** [comment]
Looks like this is upstream and will be fixed in TypeScript 4.3.0 (see: https://github.com/microsoft/TypeScript/issues/33726).

**Comment by kitsonk** [comment]
Ref: #9960

**Comment by bartlomieju** [comment]
TypeScript 4.3 landed in #9960, so closing as resolved.
